src/saturn_engine/worker/topics/logger.py

```diff
@@ -0,0 +1,56 @@
+from typing import Any
+from typing import AsyncGenerator
+from typing import Callable
+from typing import Optional
+
+import dataclasses
+import json
+from enum import Enum
+
+from saturn_engine.core import TopicMessage
+from saturn_engine.utils.log import getLogger
+from saturn_engine.worker.topic import Topic
+
+
+class LoggingTopic(Topic):
+    """A topic contains default loggers for error handling
+    such as ignoring the error and logging it in debug.
+    """
+
+    class Channels(Enum):
+        IGNORE = "ignore"
+        DEBUG = "debug"
+        INFO = "info"
+        WARNING = "warning"
+        ERROR = "error"
+
+    @dataclasses.dataclass
+    class Options:
+        channel: Optional[str] = None
+
+    def resolve_channel_to_fn(self) -> Callable[[Any], None]:
+        if self.channel == self.Channels.IGNORE.value:
+            return lambda *args: None
+        elif self.channel == self.Channels.DEBUG.value:
+            return self.logger.debug
+        elif self.channel == self.Channels.INFO.value:
+            return self.logger.info
+        elif self.channel == self.Channels.WARNING.value:
+            return self.logger.warning
+        elif self.channel == self.Channels.ERROR.value:
+            return self.logger.error
+        else:
+            raise ValueError(f"Unknown channel: {self.channel}")
+
+    def __init__(self, options: Options, **kwargs: object) -> None:
+        self.channel = options.channel or self.Channels.IGNORE.value
+        self.logger = getLogger(__name__, self)
+        self.logger_fn: Callable[[Any], None] = self.resolve_channel_to_fn()
+
+    async def run(self) -> AsyncGenerator[TopicMessage, None]:
+        return
+        yield
+
+    async def publish(self, message: TopicMessage, wait: bool) -> bool:
+        self.logger_fn(json.dumps({"id": message.id, "args": message.args}))
```

Comment:
structured data must go to `extra` argument (Check elsewhere in the code). Keep message human readable.
Since the main use case of this is to dump exception, I would do the extra work to check if the message is an error message and format with a backtrace and all (And pass it to extra through `exception`, check the logger service, there's a way exception are extracted from sys.exc_info and pass to extra.

Reply:
I can't seem to find any reference to pass "exception" inside extra. 

We could however do the following:

```python
async def publish(self, message: TopicMessage, wait: bool) -> bool:
    if self.options.level is None:
        return True
    
    extra = {"data":{"id": message.id, "args": message.args}}
    
    exc_info = None
    status = "success"
    if "traceback" in message.args:
        status = "error"
        exc_info = sys.exc_info()

    self.logger.log(
        self.options.level.value,
        f"Message executed with {status}",
        extra=extra,
        exc_info=exc_info,
    )
    return True
```

wdyt?

Reply:
Right, it seems to be logger specific. In that case I would dump the backtrace of the error into `extra={"data": {"exception": backtrace}}`, I think it should give something similar to what `.exception` would give.

Reply:
And as the message, `f{message.args["type"]}: f{message.args["message"]}`

Reply:
> Right, it seems to be logger specific. In that case I would dump the backtrace of the error into `extra={"data": {"exception": backtrace}}`, I think it should give something similar to what `.exception` would give.

Just to clear, I mean: `backtrace = message.args["traceback"].format()`. Also make sure to only do this when it's an error message! Otherwise, might just want to log the message id and pass the message args in extra.data.